src/core/orderbook.js

```
@@ -933,41 +933,44 @@ OrderBook.prototype.insertOffer = function(node) {
+  try {
+    const offer = OrderBook.offerRewrite(node.fields);
+    const takerGets = this.normalizeAmount(this._currencyGets, offer.TakerGets);
+    const takerPays = this.normalizeAmount(this._currencyPays, offer.TakerPays);
 
-  offer.LedgerEntryType = node.entryType;
-  offer.index = node.ledgerIndex;
+    offer.LedgerEntryType = node.entryType;
+    offer.index = node.ledgerIndex;
 
-  // We're safe to calculate quality for newly created offers
-  offer.quality = takerPays.divide(takerGets).to_text();
+    // We're safe to calculate quality for newly created offers
+    offer.quality = takerPays.divide(takerGets).to_text();
 
-  const originalLength = this._offers.length;
+    const originalLength = this._offers.length;
 
-  for (let i = 0; i < originalLength; i++) {
-    const quality = OrderBookUtils.getOfferQuality(offer, this._currencyGets);
-    const existingOfferQuality = OrderBookUtils.getOfferQuality(
-      this._offers[i],
-      this._currencyGets
-    );
+    for (let i = 0; i < originalLength; i++) {
+      const quality = OrderBookUtils.getOfferQuality(offer, this._currencyGets);
+      const existingOfferQuality = OrderBookUtils.getOfferQuality(
+        this._offers[i],
+        this._currencyGets
+      );
 
-    if (quality.compareTo(existingOfferQuality) <= 0) {
-      this._offers.splice(i, 0, offer);
+      if (quality.compareTo(existingOfferQuality) <= 0) {
+        this._offers.splice(i, 0, offer);
 
-      break;
+        break;
+      }
     }
-  }
 
-  if (this._offers.length === originalLength) {
-    this._offers.push(offer);
-  }
-
-  this.incrementOwnerOfferCount(offer.Account);
-
-  this.updateOwnerOffersFundedAmount(offer.Account);
+    if (this._offers.length === originalLength) {
+      this._offers.push(offer);
+    }
 
-  this.emit('offer_added', offer);
+    this.incrementOwnerOfferCount(offer.Account);
+    this.updateOwnerOffersFundedAmount(offer.Account);
+    this.emit('offer_added', offer);
+  } catch (e) {
```

Personally I'd rather something blow up (fail fast) than try and continue on after these sorts of errors.  Wouldn't it be better to fix the bug?